hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1413,18 +1422,26 @@ private static List<String> getColumnsToIndex(HoodieTableConfig tableConfig,
    */
   private static Stream<String> getColumnsToIndexWithoutRequiredMetaFields(HoodieMetadataConfig metadataConfig,
                                                                            Either<List<String>, Lazy<Option<Schema>>> tableSchema,
+                                                                           boolean freshTable,
                                                                            Option<HoodieRecordType> recordType) {
     List<String> columnsToIndex = metadataConfig.getColumnsEnabledForColumnStatsIndex();
     if (!columnsToIndex.isEmpty()) {
+      if (freshTable) {
```

Comment:
for a very fresh table, we may not have table schema to validate against. So, just honor the columns to index overriden. Subsequent commit will auto correct the list of cols to index if need be,